### PR TITLE
Feature/seriestags api

### PR DESCRIPTION
**Pull request by kobmic**

### CODE REVIEW

File: src/main/scala/se/nimsa/sbx/app/routing/SeriesTagsRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                complete(seriesTags)
+            }
+
+          } ~ post {
+            entity(as[SeriesTag]) { seriesTag =>
+              onComplete(metaDataService.ask(CreateSeriesTag(seriesTag))) {
+                case Success(tag: SeriesTag) => complete((Created, tag))
+                case Failure(ex) => {
+                  val msg = ex.getMessage
+                  if (msg.contains("idx_unique_series_tag_name")) {
```
Comment (KarlSjostrand) [inline src/main/scala/se/nimsa/sbx/app/routing/SeriesTagsRoutes.scala]: This seems a little too "stringly typed". Should we introduce a new Exception? Or throw an IllegalArgumentException which is always mapped to BadRequest.

File: src/main/scala/se/nimsa/sbx/app/routing/SeriesTagsRoutes.scala [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+
+// POST /metadata/series/{id}/seriestags
+// DEL  /metadata/series/{seriesId}/seriestags/{seriesTagId}
+
+// GET  /metadata/seriestags
+
+
+
+// TODO:
```
Comment (KarlSjostrand) [inline src/main/scala/se/nimsa/sbx/app/routing/SeriesTagsRoutes.scala]: Isn't this implemented below? Remove these comments?

File: src/main/scala/se/nimsa/sbx/metadata/MetaDataServiceActor.scala [hunk 3]
```diff
@@ -60,6 +60,18 @@ class MetaDataServiceActor(metaDataDao: MetaDataDAO, propertiesDao: PropertiesDA
         case GetSeriesTags =>
           pipe(propertiesDao.listSeriesTags.map(SeriesTags)).to(sender)
 
+        case GetSeriesTag(tagId) =>
+          pipe(propertiesDao.seriesTagForId(tagId)).to(sender)
+
+        case UpdateSeriesTag(tag) =>
```
Comment (KarlSjostrand) [inline src/main/scala/se/nimsa/sbx/metadata/MetaDataServiceActor.scala]: The current pattern in slicebox for protecting the database from concurrent updates is to use `pipeSequentiallyTo` for anything that changes the database (update, create and delete here). Mix in the trait `SequentialPipeToSupport` to get this functionality.

File: src/main/scala/se/nimsa/sbx/metadata/PropertiesDAO.scala [hunk 4]
```diff
@@ -251,6 +251,7 @@ class PropertiesDAO(val dbConf: DatabaseConfig[JdbcProfile])(implicit ec: Execut
       }
   }
 
+  // FIXME
```
Comment (KarlSjostrand) [inline src/main/scala/se/nimsa/sbx/metadata/PropertiesDAO.scala]: Is this fixed?

File: src/test/scala/se/nimsa/sbx/metadata/PropertiesDAOTest.scala [hunk 5]
```diff
@@ -800,8 +806,37 @@ class PropertiesDAOTest extends AsyncFlatSpec with Matchers with BeforeAndAfterA
...
+      (_, (_, _), (dbSeries1, dbSeries2, dbSeries3, _), (_, _, _, _, _, _, _, _)) <- insertMetaDataAndProperties()
+      seriesTags1 <- propertiesDao.listSeriesTags
+      _ <- propertiesDao.deleteSeriesTag(seriesTags1.head.id)
+      seriesTags2 <- propertiesDao.listSeriesTags
+    } yield {
+      seriesTags1.size should be(2)
+      seriesTags1.map(_.name) should be(List("Tag1", "Tag2"))
+      seriesTags2.size should be(1)
+      seriesTags2.map(_.name).exists(_ == "Tag1") shouldBe false
     }
```
Comment (KarlSjostrand) [inline src/test/scala/se/nimsa/sbx/metadata/PropertiesDAOTest.scala]: Are we missing tests for create series tag or is this part of the other tests?

File: src/main/scala/se/nimsa/sbx/metadata/MetaDataServiceActor.scala (hunk 3 above)
Comment (kobmic) [reply src/main/scala/se/nimsa/sbx/metadata/MetaDataServiceActor.scala]: fixed

File: src/test/scala/se/nimsa/sbx/metadata/PropertiesDAOTest.scala (hunk 5 above)
Comment (kobmic) [reply src/test/scala/se/nimsa/sbx/metadata/PropertiesDAOTest.scala]: "create" is tested implicit through insertMetaDataAndProperties() which is used in all tests

File: src/main/scala/se/nimsa/sbx/app/routing/SeriesTagsRoutes.scala (hunk 1 above)
Comment (kobmic) [reply src/main/scala/se/nimsa/sbx/app/routing/SeriesTagsRoutes.scala]: found no better solution, exception is jdbc driver specific

**Pull request merged**